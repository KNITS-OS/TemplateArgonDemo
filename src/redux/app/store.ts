import { configureStore } from "@reduxjs/toolkit";
import counterReducer from "redux/features/example/counterSlice";
import sidenavReducer from "redux/features/sidenav/sidenavSlice";
import { employeesApiSlice } from "../features/employees/employeesApiSlice";
import { groupsApiSlice } from "../features/groups/groupsApiSlice";

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    sidenav: sidenavReducer,
    [employeesApiSlice.reducerPath]: employeesApiSlice.reducer,
    [groupsApiSlice.reducerPath]: groupsApiSlice.reducer,
  },
  // add special capabilites to the store
  // like tracking cache lifetimes
  // if no other part of the codebase needs this data
  // it will be removed from the cache after it expires
  middleware: getDefaultMiddleware => {
    return getDefaultMiddleware().concat(
      employeesApiSlice.middleware,
      groupsApiSlice.middleware,
    );
  },
});

export type RootState = ReturnType<typeof store.getState>;

export type AppDisptatch = typeof store.dispatch;
