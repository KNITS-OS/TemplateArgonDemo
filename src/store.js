import { applyMiddleware, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import logger from "redux-logger";
import thunk from "redux-thunk";
import { rootReducer } from "redux/root.reducer.js";

const middleware = [thunk];

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(...middleware, logger)),
);

export default store;
