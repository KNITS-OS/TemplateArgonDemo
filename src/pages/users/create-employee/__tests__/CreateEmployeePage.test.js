import { Provider } from "react-redux";

import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { store } from "store";

import { CreateEmployeePage } from "..";

const MockCreateGroupsPage = () => {
  return (
    <Provider store={store}>
      <CreateEmployeePage />
    </Provider>
  );
};

describe("Create Employees Page", () => {
  test("should write into input, see that this text was written and is on the screen", () => {
    render(<MockCreateGroupsPage />);

    // example with testid
    // const inputElement = screen.getByTestId(/firstName/i);
    const inputElement = screen.getByPlaceholderText(/First Name/i);
    const inputValue = "Hamli";

    // https://polvara.me/posts/five-things-you-didnt-know-about-testing-library
    // nr 5 explains why userEvent is better
    userEvent.type(inputElement, inputValue);

    // example with fireevent
    // fireEvent.change(inputElement, { target: { value: "Hamli" } });
    expect(inputElement.value).toBe(inputValue);
    expect(screen.getByDisplayValue("Hamli")).toBeInTheDocument();
  });
  describe("Snapshot tests", () => {
    test("should render correctly", () => {
      const { asFragment } = render(<MockCreateGroupsPage />);

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
