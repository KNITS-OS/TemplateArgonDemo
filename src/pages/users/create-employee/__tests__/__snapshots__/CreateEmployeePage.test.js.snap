// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Employees Page Snapshot tests should render correctly 1`] = `
<DocumentFragment>
  <div
    class="header pb-6 d-flex align-items-center"
    style="min-height: 150px; background-size: cover; background-position: center top;"
  >
    <span
      class="mask bg-gradient-info opacity-8"
    />
  </div>
  <div
    class="mt--6 container-fluid"
  >
    <div
      class="row"
    >
      <div
        class="order-xl-1 col-xl-12"
      >
        <div
          class="card"
        >
          <div
            class="card-header"
          >
            <div
              class="align-items-center row"
            >
              <div
                class="col-8"
              >
                <h3
                  class="mb-0"
                >
                  Create Employee
                </h3>
              </div>
            </div>
          </div>
          <div
            class="card-body"
          >
            <h6
              class="heading-small text-muted mb-4"
            >
              User information
            </h6>
            <div
              class="pl-lg-4"
            >
              <div
                class="row"
              >
                <div
                  class="col-lg-6"
                >
                  <div
                    class="form-group"
                  >
                    <label
                      class="form-control-label"
                      for="input-first-name"
                    >
                      First name
                    </label>
                    <input
                      class="form-control"
                      id="input-first-name"
                      label="First name"
                      placeholder="First name"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="col-lg-6"
                >
                  <div
                    class="form-group"
                  >
                    <label
                      class="form-control-label"
                      for="input-last-name"
                    >
                      Last Name
                    </label>
                    <input
                      class="form-control"
                      id="input-last-name"
                      label="Last Name"
                      placeholder="Last Name"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col-lg-6"
                >
                  <div
                    class="form-group"
                  >
                    <label
                      class="form-control-label"
                      for="input-international-name"
                    >
                      International Name
                    </label>
                    <input
                      class="form-control"
                      id="input-international-name"
                      label="International Name"
                      placeholder="International Name"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="col-lg-6"
                >
                  <div
                    class="form-group"
                  >
                    <label
                      class="form-control-label"
                      for="input-email"
                    >
                      Email Address
                    </label>
                    <input
                      class="form-control"
                      id="input-email"
                      label="Email Address"
                      placeholder="Email Address"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <hr
              class="my-4"
            />
            <h6
              class="heading-small text-muted mb-4"
            >
              Contact information
            </h6>
            <div
              class="pl-lg-4"
            >
              <div
                class="row"
              >
                <div
                  class="col-md-12"
                >
                  <div
                    class="form-group"
                  >
                    <label
                      class="form-control-label"
                      for="input-address"
                    >
                      Office Street
                    </label>
                    <input
                      class="form-control"
                      id="input-address"
                      label="Office Street"
                      placeholder="Office Street"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col-lg-4"
                >
                  <div
                    class="form-group"
                  >
                    <label
                      class="form-control-label"
                      for="input-city"
                    >
                      Office City
                    </label>
                    <input
                      class="form-control"
                      id="input-city"
                      label="Office City"
                      placeholder="Office City"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="col-lg-4"
                >
                  <div
                    class="form-group"
                  >
                    <label
                      class="form-control-label"
                      for="input-country"
                    >
                      Office Country
                    </label>
                    <input
                      class="form-control"
                      id="input-country"
                      label="Office Country"
                      placeholder="Office Country"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="col-lg-4"
                >
                  <div
                    class="form-group"
                  >
                    <label
                      class="form-control-label"
                      for="input-postal-code"
                    >
                      Office Postal code
                    </label>
                    <input
                      class="form-control"
                      id="input-postal-code"
                      label="Office Postal code"
                      placeholder="Office Postal code"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <hr
              class="my-4"
            />
            <h6
              class="heading-small text-muted mb-4"
            >
              Company Data
            </h6>
            <div
              class="pl-lg-4"
            >
              <div
                class="row"
              >
                <div
                  class="col-lg-4"
                >
                  <div
                    class="form-group"
                  >
                    <label
                      class="form-control-label"
                      for="input-title"
                    >
                      Job Title
                    </label>
                    <input
                      class="form-control"
                      id="input-title"
                      label="Job Title"
                      placeholder="Job Title"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="col-lg-4"
                >
                  <div
                    class="form-group"
                  >
                    <label
                      class="form-control-label"
                      for="input-company-phone"
                    >
                      Company Phone
                    </label>
                    <input
                      class="form-control"
                      id="input-company-phone"
                      label="Company Phone"
                      placeholder="Company Phone"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="col-lg-4"
                >
                  <div
                    class="form-group"
                  >
                    <label
                      class="form-control-label"
                      for="input-company-code"
                    >
                      Company Code
                    </label>
                    <input
                      class="form-control"
                      id="input-company-code"
                      label="Company Code"
                      placeholder="Company Code"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col-lg-4"
                >
                  <div
                    class="form-group"
                  >
                    <label
                      class="form-control-label"
                      for="input-business-unit"
                    >
                      Business Unit
                    </label>
                    <input
                      class="form-control"
                      id="input-business-unit"
                      label="Business Unit"
                      placeholder="Business Unit"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="col-lg-4"
                >
                  <div
                    class="form-group"
                  >
                    <label
                      class="form-control-label"
                      for="input-cost-center"
                    >
                      Cost Center
                    </label>
                    <input
                      class="form-control"
                      id="input-cost-center"
                      label="Cost Center"
                      placeholder="Cost Center"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="col-lg-4"
                >
                  <div
                    class="form-group"
                  >
                    <label
                      class="form-control-label"
                      for="input-management-group"
                    >
                      Management Group
                    </label>
                    <input
                      class="form-control"
                      id="input-management-group"
                      label="Management Group"
                      placeholder="Management Group"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="align-items-center py-4 row"
              >
                <div
                  class="text-right col-7 col-lg-12"
                >
                  <button
                    class="btn btn-success"
                    type="button"
                  >
                    Submit
                  </button>
                   
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
